presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -171,6 +171,11 @@
     void beginQuery(Session session, String catalogName);
 
     /**
+     * Ends query. This is the very last notification when query is done.
```

Comment:
Suggest: Ends the query. This is the very last notification after the query finishes, whether it succeeds or fails.